force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +150,28 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
+        String filteredNames = '';
 
         // Loop through the list of scientists' names
-
+        for (String name : scientistNames) {
             // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
+            if (name.containsIgnoreCase('G')) {
+                filteredNames += name + ',';
+            } else {
+                continue;
+            }
+
+            // if (name.toLowerCase().containsNone('g')) {
+            //     continue;
+            // } else {
+            //     // Add the name to the result string, followed by a comma
+            //     filteredNames += name + ',';
+            // }
+        }
         // Remove the last comma from the result string
+        filteredNames = filteredNames.replaceAll(',$', '');
```

Comment:
I'm not sure what's going on here. A method to consider in the future is `removeEnd()`:
```suggestion
        filteredNames = filteredNames.removeEnd(',');
```

Reply:
That's nice... ya I used some crazy regex, I didn't realize apex had that method. Way more elegant.  should have looked at the string methods better thanks!